### ISSUE TITLE
[Retail Bug Report] Compatibility issue with SilverDragon

### ISSUE THREAD

**Issue by AdamXA** [tukui-org/ElvUI#119]
### What is the issue you are having?

SilverDragon starts erroring while trying to render world map overlays.

### What is the expected behavior?

There should be routes showing the paths of NPCs on the world map.

### What actually happened?

I am getting a lua error when using ElvUI 12.58

This bug was introduced in either 12.57 (guessing this one because there was world map fixes in it) or 12.58.

Works fine in 12.56. I have downgraded ElvUI for now, to avoid this issue.

### Errors

```shell
Message: Interface\SharedXML\MathUtil.lua:129: attempt to perform arithmetic on local 'y2' (a nil value)
Time: Wed Dec  8 15:30:38 2021
Count: 1
Stack: Interface\SharedXML\MathUtil.lua:129: attempt to perform arithmetic on local 'y2' (a nil value)
[string "=[C]"]: ?
[string "@Interface\SharedXML\MathUtil.lua"]:129: in function <Interface\SharedXML\MathUtil.lua:128>
[string "=(tail call)"]: ?
[string "@Interface\AddOns\SilverDragon_Overlay\worldmap.lua"]:158: in function `Connect'
[string "@Interface\AddOns\SilverDragon_Overlay\worldmap.lua"]:109: in function `ConnectPins'
[string "@Interface\AddOns\SilverDragon_Overlay\worldmap.lua"]:95: in function `AddRoute'
[string "@Interface\AddOns\SilverDragon_Overlay\worldmap.lua"]:66: in function `RefreshAllData'
[string "@Interface\AddOns\Blizzard_MapCanvas\MapCanvas_DataProviderBase.lua"]:76: in function `OnMapChanged'
[string "@Interface\AddOns\Blizzard_MapCanvas\Blizzard_MapCanvas.lua"]:580: in function <...ace\AddOns\Blizzard_MapCanvas\Blizzard_MapCanvas.lua:579>
[string "=[C]"]: in function `secureexecuterange'
[string "@Interface\AddOns\Blizzard_MapCanvas\Blizzard_MapCanvas.lua"]:583: in function `OnMapChanged'
[string "@Interface\AddOns\Blizzard_WorldMap\Blizzard_WorldMap.lua"]:234: in function `OnMapChanged'
...
[string "=[C]"]: in function `Show'
[string "@Interface\FrameXML\UIParent.lua"]:2958: in function `SetUIPanel'
[string "@Interface\FrameXML\UIParent.lua"]:2764: in function `ShowUIPanel'
[string "@Interface\FrameXML\UIParent.lua"]:2671: in function <Interface\FrameXML\UIParent.lua:2667>
[string "=[C]"]: in function `SetAttribute'
[string "@Interface\FrameXML\UIParent.lua"]:3522: in function `ShowUIPanel'
[string "@Interface\FrameXML\QuestLogOwnerMixin.lua"]:103: in function `SetDisplayState'
[string "@Interface\FrameXML\QuestLogOwnerMixin.lua"]:41: in function `HandleUserActionToggleSelf'
[string "@Interface\AddOns\Blizzard_WorldMap\Blizzard_WorldMap.lua"]:440: in function `ToggleWorldMap'
[string "TOGGLEWORLDMAP"]:1: in function <[string "TOGGLEWORLDMAP"]:1>
```

### ElvUI Changelog/Troubleshooting

- [X] I have confirmed that I have read the [Changelog](https://www.tukui.org/download.php?ui=elvui&changelog) and done the [Troubelshooting Steps](https://www.tukui.org/forum/viewtopic.php?p=28#p28).

**Comment by Azilroka** [comment]
Nope. We have changed nothing we dont change the world map.

**Comment by Azilroka** [comment]
![image](https://user-images.githubusercontent.com/1934853/145293650-414dfa7d-8b28-4231-9f35-b5b782b871a9.png)

**Comment by Azilroka** [comment]
![image](https://user-images.githubusercontent.com/1934853/145293706-6199fe8f-8e5a-4ece-8ac0-3d9ec0611314.png)

**Comment by AdamXA** [comment]
Hi can you try opening the storm peaks map with it open?

**Comment by AdamXA** [comment]
> Nope. We have changed nothing we dont change the world map.

https://github.com/tukui-org/ElvUI/commits/705798621bf390fbf134c59c4bf2aea3bb9ca34b/ElvUI/Core/Modules/Maps/Worldmap.lua

**Comment by AdamXA** [comment]
![image](https://user-images.githubusercontent.com/29933042/145393377-1812fdde-f582-4c96-9944-b10e85109627.png)

**Comment by AdamXA** [comment]
Only ElvUI and Silverdragon on

**Comment by Azilroka** [comment]
> > Nope. We have changed nothing we dont change the world map.
> 
> https://github.com/tukui-org/ElvUI/commits/705798621bf390fbf134c59c4bf2aea3bb9ca34b/ElvUI/Core/Modules/Maps/Worldmap.lua

https://github.com/tukui-org/ElvUI/compare/v12.56...main

**Comment by Azilroka** [comment]
`Message: Interface\SharedXML\MathUtil.lua:129: attempt to perform arithmetic on local 'y2' (a nil value)
Time: Wed Dec  8 15:30:38 2021
Count: 1
Stack: Interface\SharedXML\MathUtil.lua:129: attempt to perform arithmetic on local 'y2' (a nil value)
[string "=[C]"]: ?
[string "@Interface\SharedXML\MathUtil.lua"]:129: in function <Interface\SharedXML\MathUtil.lua:128>
[string "=(tail call)"]: ?
[string "@Interface\AddOns\SilverDragon_Overlay\worldmap.lua"]:158: in function `Connect'
[string "@Interface\AddOns\SilverDragon_Overlay\worldmap.lua"]:109: in function `ConnectPins'
[string "@Interface\AddOns\SilverDragon_Overlay\worldmap.lua"]:95: in function `AddRoute'
[string "@Interface\AddOns\SilverDragon_Overlay\worldmap.lua"]:66: in function `RefreshAllData'
[string "@Interface\AddOns\Blizzard_MapCanvas\MapCanvas_DataProviderBase.lua"]:76: in function `OnMapChanged'
[string "@Interface\AddOns\Blizzard_MapCanvas\Blizzard_MapCanvas.lua"]:580: in function <...ace\AddOns\Blizzard_MapCanvas\Blizzard_MapCanvas.lua:579>
[string "=[C]"]: in function `secureexecuterange'
[string "@Interface\AddOns\Blizzard_MapCanvas\Blizzard_MapCanvas.lua"]:583: in function `OnMapChanged'
[string "@Interface\AddOns\Blizzard_WorldMap\Blizzard_WorldMap.lua"]:234: in function `OnMapChanged'`

No mention of ElvUI.

**Comment by AdamXA** [comment]
> Now you want me to proceed to another area. I will do that.

Yea sorry about that, only realized that it was only reproducible there from trying it later when I was not in that zone.

I have also noticed you need to login/reload in the zone to reproduce it reliably

**Comment by kodewdle** [comment]
uncheck 
![image](https://user-images.githubusercontent.com/2266754/145478305-4d83d2ce-aaa0-4f34-a43c-bc3ac940d7e0.png)

**Comment by AdamXA** [comment]
> uncheck ![image](https://user-images.githubusercontent.com/2266754/145478305-4d83d2ce-aaa0-4f34-a43c-bc3ac940d7e0.png)

Thank you, this prevents the error in 12.58!

**Comment by kodewdle** [comment]
closed with 41fb87402fd747644a062d0c36a773875ee8ad09 - retail doesnt seem to need the attributes
